### PR TITLE
generic EoE stack parts

### PR DESCRIPTION
**Pull request by nakarlsson**
Application sample for lwIP will be added later

### CODE REVIEW

**Comment by nakarlsson** [comment]
I've tested with ping & netio via TwinCAT and lwIP on target.  Tested layer 2 and getIP & setIP via SOEM hack.

Mbx size 128 to force fragmentation

--------------------------------------------------------------------------
D:\Project\LAM\SOES\soes_ws\rt-kernel\test\netio\windows>win32-i386.exe -b 1000 -t 192.168.9.200

NETIO - Network Throughput Benchmark, Version 1.26
(C) 1997-2005 Kai Uwe Rommel

TCP connection established.
Packet size 1000 bytes:  63833 Byte/s Tx,  19308 Byte/s Rx.
Done.
--------------------------------------------------------------------------
D:\Project\LAM\SOES\soes_ws\rt-kernel\test\netio\windows>ping -l 1024 192.168.9.200

Pinging 192.168.9.200 with 1024 bytes of data:
Reply from 192.168.9.200: bytes=1024 time=64ms TTL=255
Reply from 192.168.9.200: bytes=1024 time=52ms TTL=255
Reply from 192.168.9.200: bytes=1024 time=51ms TTL=255
Reply from 192.168.9.200: bytes=1024 time=52ms TTL=255

Ping statistics for 192.168.9.200:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 51ms, Maximum = 64ms, Average = 54ms

--------------------------------------------------------------------------

**Comment by nakarlsson** [comment]
Added eoe to cmake systems, ( forgot it before )
Also aded a more consistant handling of the IP address, the eoe_ip4_addr stor the address in network order. Input/Output uint32 is in host order as one might expect

**Comment by nakarlsson** [comment]
ping @ArthurKetels @hefloryd

File: soes/esc_eoe.c [hunk 1]
```diff
@@ -0,0 +1,1020 @@
...
+   char dns_name[EOE_DNS_NAME_LENGTH];
+} eoe_param_t;
+
+/** Main EoE status data array. Structure gets filled with current information
+ * variables during EoE receive and send operations.
+ */
+static _EOEvar EOEvar;
+
+/** Main FoE configuration pointer data array. Structure is allocated and filled
+ * by the application defining what preferences it require.
```
Comment (ArthurKetels) [inline soes/esc_eoe.c]: require -> requires

File: soes/esc_eoe.c [hunk 2]
```diff
@@ -0,0 +1,1020 @@
...
+   data_offset = EOE_PARAM_OFFSET;
+
+   if(port  > EOE_NUMBER_OF_PORTS)
+   {
+      DPRINT("Invalid port\n");
+      port = 0;
+   }
+   else
+   {
+      port = 0;
```
Comment (ArthurKetels) [inline soes/esc_eoe.c]: Is this intended?

**Comment by ArthurKetels** [comment]
Looks OK. Test application is needed though.

**Comment by nakarlsson** [comment]
I've added a prototype application, so the PR no consist of two commits. Do a hard reset or re-clone my branch to get the latest.

**Comment by ArthurKetels** [comment]
Is see no need to delay this. Just merge it as well as the SOEM part.

**Pull request merged**